centreon/centreon-stream-connector-scripts
# (stream/lib): sc_flush - refacto queue system and compatibility patch for apiv2 stream connectors

This patch needs a full test from multiple people

purpose of this PR: 

make sure stream connectors make the best use of the centreon broker retention system

what it does: 

- only return true when a queue is successfully flushed. Meaning that broker can also acknowledge events
- when an event is stored in a queue, return false until queue is flushed
- add a flush() method that is called by broker every now and then. This will try to flush all queue if last global flush was at least 5 minutes ago. this will allow broker to acknowledge events on a regular basis 

how can it be tested: 

- you can reduce the **Event queue max size** in your broker configuration to create retention faster if you do not have a lot of services on your centreon platform
- configure a refactored stream connector (use service now or elastic because they can send bulked events) so it sends events (it needs to fail so configure a fake address)
- broker needs to create retention
- change the configuration of your stream connector by adding the send_data_test parameter 
- retention should be cleared because events are going to acknowledged since they are successfully sent in the logfile

repeat above tests by having a max_buffer_size of 1 and a max_buffer_size above 1.
